From: Didier Roche <didrocks@ubuntu.com>
Date: Tue, 7 Aug 2018 11:15:57 +0100
Subject: Prefer ubuntu session as fallback over GNOME

As GNOME, optionally installed, is a Wayland session and ubuntu
is now transitioned as a Xorg session, the fallback will prefer
naturally the GNOME session. Ensure we prioritize the ubuntu
session in fallback order.
Forwarded: not-needed
---
 daemon/gdm-session.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/daemon/gdm-session.c b/daemon/gdm-session.c
index e1a46c8..d5e0ae7 100644
--- a/daemon/gdm-session.c
+++ b/daemon/gdm-session.c
@@ -578,6 +578,14 @@ get_fallback_session_name (GdmSession *self)
                 }
         }
 
+        name = g_strdup ("ubuntu");
+        if (get_session_command_for_name (self, name, NULL)) {
+                g_free (self->fallback_session_name);
+                self->fallback_session_name = name;
+                goto out;
+        }
+        g_free (name);
+
         name = g_strdup ("gnome");
         if (get_session_command_for_name (self, name, NULL, NULL)) {
                 g_free (self->fallback_session_name);
